### PR TITLE
Replace deprecated libsass with dart-sass

### LINKED ISSUE #44: Replace node-sass with dart-sass(upstream deprecated)
See https://github.com/sass/node-sass#node-sass

### PR DESCRIPTION
**Pull request by robincafolla**
Fixes #44 

@knksmith57 @xtexChooser

### CODE REVIEW

**Comment by robincafolla** [comment]
Should also fix #43

**Comment by xtexChooser** [comment]
@knksmith57 Come here

**Comment by tufu9441** [comment]
Hope to be merged soon.

**Comment by robincafolla** [comment]
Hey @knksmith57. It has been a while since I wrote the PR, but I'll try answer these:

1. I may have to check to be sure, but I don't think the change in `outputStyle` was a preference; I think I needed to make that change to get it to work.
2. Great question. I imagine it could mess up config loaded for another library using the `sass` key. If you make that change I'd probably at least tick the major version number and make it clear in the change notes what's gone on. That said, given that the library conflicts on node 16+ (the current LTS) it's probably fairly safe.
3. Happy to help. I use hexo for 3 or 4 sites atm, but I actually don't use much sass :sweat_smile:. If the maintenance burden remains low I'm happy to help out.

No worries. I know how it is.

**Comment by tufu9441** [comment]
I have been using https://github.com/cocowool/hexo-renderer-sass to solve the problem temporarily before this PR is merged, and nothing seems to have gone wrong.

**Comment by robincafolla** [comment]
@knksmith57 tested the nested vs expanded outputStyle.

The nested output style doesn't seem to be supported in dart-sass. The option isn't listed in the [docs](https://sass-lang.com/documentation/js-api/modules#OutputStyle) and passing it throws `Error: Invalid argument(s): Unsupported output style "nested".`.

**Comment by robincafolla** [comment]
@knksmith57 need anything more from me on this? Otherwise, I think it's safe to merge.

**Comment by kmuncie** [comment]
As I still use Hexo for my personal blog I have forked this project to try and update and maintain it down the road https://github.com/kmuncie/hexo-renderer-sass - Thanks @knksmith57 for the original work, I have benefited from it for many years!

**Comment by Neats29** [comment]
> As I still use Hexo for my personal blog I have forked this project to try and update and maintain it down the road https://github.com/kmuncie/hexo-renderer-sass - Thanks @knksmith57 for the original work, I have benefited from it for many years!

Thank you so much for making this fork, lifesaver 🙏

File: lib/renderer.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = (ext) => function (data) {
   var config = extend({
     data: data.text,
     file: data.path,
-    outputStyle: 'nested',
+    outputStyle: 'expanded',
     sourceComments: false,
```
Comment (JLHwung) [inline lib/renderer.js:17]:
```suggestion
```
Per https://github.com/sass/dart-sass#legacy-javascript-api, the `dartSass` does not support `sourceComments` anyway.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hexo-renderer-sass",
-  "version": "0.4.0",
+  "version": "0.5.0",
```
Comment (JLHwung) [inline package.json:3]:
```suggestion
  "version": "0.4.0",
```
The expected version will be bumped when I run `npm version major`.

**Review by JLHwung — APPROVED** [review]
Sorry for the super late reply. The changes look good to me. @robincafolla Awesome work, thank you!

**Comment by xtexChooser** [comment]
well thanks for update!

**Comment by robincafolla** [comment]
No worries. Glad to help.

**Pull request merged**